pkg/protocol/sybilprotection/sybilprotection.go

```diff
@@ -21,13 +21,13 @@ type SybilProtection interface {
 	// The first epoch in which rewards existed is returned (firstRewardEpoch).
 	// Since the validator may still be active and the EndEpoch might be in the future, the epoch until which rewards were calculated is returned in addition to the first epoch in which rewards existed (lastRewardEpoch).
 	// The rewards are decayed until claimingEpoch, which should be set to the epoch in which the rewards would be claimed.
-	ValidatorReward(validatorID iotago.AccountID, stakingFeature *iotago.StakingFeature, claimingEpoch iotago.EpochIndex) (validatorReward iotago.Mana, firstRewardEpoch iotago.EpochIndex, lastRewardEpoch iotago.EpochIndex, err error)
+	ValidatorReward(validatorID iotago.AccountID, stakingFeature *iotago.StakingFeature, claimingEpoch iotago.EpochIndex, retentionPeriod iotago.EpochIndex) (validatorReward iotago.Mana, firstRewardEpoch iotago.EpochIndex, lastRewardEpoch iotago.EpochIndex, err error)
 	// DelegatorReward returns the amount of mana that a delegator has earned in a given epoch range.
 	//
 	// The first epoch in which rewards existed is returned (firstRewardEpoch).
 	// Since the Delegation Output's EndEpoch might be unset due to an ongoing delegation, the epoch until which rewards were calculated is also returned (lastRewardEpoch).
 	// The rewards are decayed until claimingEpoch, which should be set to the epoch in which the rewards would be claimed.
-	DelegatorReward(validatorID iotago.AccountID, delegatedAmount iotago.BaseToken, epochStart iotago.EpochIndex, epochEnd iotago.EpochIndex, claimingEpoch iotago.EpochIndex) (delegatorReward iotago.Mana, firstRewardEpoch iotago.EpochIndex, lastRewardEpoch iotago.EpochIndex, err error)
+	DelegatorReward(validatorID iotago.AccountID, delegatedAmount iotago.BaseToken, epochStart iotago.EpochIndex, epochEnd iotago.EpochIndex, claimingEpoch iotago.EpochIndex, retentionPeriod iotago.EpochIndex) (delegatorReward iotago.Mana, firstRewardEpoch iotago.EpochIndex, lastRewardEpoch iotago.EpochIndex, err error)
```

Comment:
Would be good to explain the behavior/effects of `retentionPeriod` in the comment. 

Another thought: why is this part of the interface and putting the "burden" on the caller. This increases the surface for errors when calling this method as the caller needs to specify the retention period. However, I think the Sybilprotection implementation should be aware of the protocol parameters already. Is there a way we can abstract it away?